### PR TITLE
Update changelog with 3.4.0 release notes

### PR DESCRIPTION
**Pull request by mejiaj**
Added 3.4.0 updates to changelogs.

Guidance changelog updates:

- Implementations #1714 
- Date picker guidance #1695
- Utilities

Component updates:

- Banner
- Header
- In-page nav
- Alert & Site Alert
- Header
- Footer
- Link
- Modal
- Step indicator

[3.4.0 Release notes ](https://github.com/uswds/uswds/releases/tag/v3.4.0)

### CODE REVIEW

File: _data/changelogs/component-date-picker.yml [hunk 1]
```diff
@@ -51,7 +58,6 @@ items:
     githubPr: 4610
     githubRepo: uswds
     versionUswds: 2.13.3
-    affectsAccessibility: true
```
Comment (mejiaj) [inline _data/changelogs/component-date-picker.yml:54]: This was a duplicate key, kept the one in line 49.

**Comment by thisisdano** [comment]
Thank you. We've got to make this part of the PR process!

File: _data/changelogs/component-alert.yml [hunk 2]
```diff
@@ -2,6 +2,13 @@ title: Alert
 type: component
 changelogURL:
 items:
+  - date: 2023-03-09
+    summary: Updated padding to accept any valid spacing token.
+    summaryAdditional: The site alert and banner component was also updated.
```
Comment (thisisdano) [inline _data/changelogs/component-alert.yml:7]:
```suggestion
    summaryAdditional:
```

File: _data/changelogs/component-alert.yml [hunk 3]
```diff
@@ -2,6 +2,13 @@ title: Alert
 type: component
 changelogURL:
 items:
+  - date: 2023-03-09
+    summary: Updated padding to accept any valid spacing token.
```
Comment (thisisdano) [inline _data/changelogs/component-alert.yml:6]:
```suggestion
    summary: Updated padding settings to accept any valid spacing token.
```

File: _data/changelogs/component-banner.yml [hunk 4]
```diff
@@ -2,6 +2,27 @@ title: Banner
...
 items:
+  - date: 2023-03-09
+    summary: Improved display of focus outline on mobile.
+    summaryAdditional: Now the focus outline is visible all around the banner on mobile devices.
+    affectsStyles: true
+    githubPr: 5013
+    githubRepo: uswds
+    versionUswds: 3.4.0
+  - date: 2023-03-09
+    summary: Updated padding to accept any valid spacing token.
```
Comment (thisisdano) [inline _data/changelogs/component-banner.yml:13]:
```suggestion
    summary: Updated padding settings to accept any valid spacing token.
```

File: _data/changelogs/component-banner.yml [hunk 5]
```diff
@@ -2,6 +2,27 @@ title: Banner
...
+    summaryAdditional: The alert and site alert component was also updated.
```
Comment (thisisdano) [inline _data/changelogs/component-banner.yml:14]:
```suggestion
    summaryAdditional:
```

File: _data/changelogs/component-footer.yml [hunk 6]
```diff
@@ -2,6 +2,15 @@ title: Footer
 type: component
 changelogURL:
 items:
+  - date: 2023-03-09
+    summary: Footer now accepts any heading level for primary link headers.
+    summaryAdditional: Use any heading level for `.usa-footer__primary-link` without it reverting to an h4 when the JavaScript rebuilds the element for mobile.
```
Comment (thisisdano) [inline _data/changelogs/component-footer.yml:7]:
```suggestion
    summaryAdditional: Use any heading level for `.usa-footer__primary-link` without it reverting to an `h4` when the JavaScript rebuilds the element for mobile.
```

File: _data/changelogs/component-in-page-navigation.yml [hunk 7]
```diff
@@ -2,6 +2,16 @@ title: In-page navigation
 type: component
 changelogURL:
 items:
+  - date: 2023-03-09
+    summary: URL in address bar is updated when navigating component.
+    summaryAdditional: Users can now see the proper anchor link in the address bar when navigating.
```
Comment (thisisdano) [inline _data/changelogs/component-in-page-navigation.yml:7]:
```suggestion
    summaryAdditional: Users can now see the proper anchor link in the address bar when navigating from the in-page navigation.
```

File: _data/changelogs/component-modal.yml [hunk 8]
```diff
@@ -2,6 +2,14 @@ title: Modal
 type: component
 changelogURL:
 items:
+- date: 2023-03-09
+  summary: Fixed Interactive form elements like Date Picker in modal window.
```
Comment (thisisdano) [inline _data/changelogs/component-modal.yml:6]:
```suggestion
  summary: Fixed the display of interactive form elements (like Date Picker) in modal windows.
```

File: _data/changelogs/component-site-alert.yml [hunk 9]
```diff
@@ -2,6 +2,13 @@ title: Site alert
 type: component
 changelogURL:
 items:
+  - date: 2023-03-09
+    summary: Updated padding to accept any valid spacing token.
```
Comment (thisisdano) [inline _data/changelogs/component-site-alert.yml:6]:
```suggestion
    summary: Updated padding settings to accept any valid spacing token.
```

File: _data/changelogs/component-site-alert.yml [hunk 10]
```diff
@@ -2,6 +2,13 @@ title: Site alert
...
+    summaryAdditional: The alert and banner component was also updated.
```
Comment (thisisdano) [inline _data/changelogs/component-site-alert.yml:7]:
```suggestion
    summaryAdditional:
```

**Review by thisisdano — APPROVED** [review]

**Comment by thisisdano** [comment]
Merging over unrelated Pa11y crashes

**Pull request merged**